d2game/d2gamescreen/main_menu.go

```diff
@@ -612,3 +620,25 @@ func (v *MainMenu) onBtnTCPIPCancelClicked() {
 func (v *MainMenu) onBtnTCPIPOkClicked() {
 	v.navigator.ToCharacterSelect(d2clientconnectiontype.LANClient, v.tcpJoinGameEntry.GetText())
 }
+
+// getLocalIP returns local machine IP address
+func (v *MainMenu) getLocalIP() string {
+	conn, err := net.Dial("udp", "8.8.8.8:80")
+
```

Comment:
This is assuming that you're on the internet. What happens when you're just on a private LAN...?

Reply:
also, `8.8.8.8:80` is magic; an assumption that the significance of this value is implicit. In 20 years, what will someone looking at this code think? just my 2 cents.

Reply:
What about `LookupHost("localhost")` or something?

Reply:
Yah, I also think, that's not a best solution, but I'm not enough in go to do it correct. If you have some time, correct it please or simply I'll remove this function and replace it with TODO

Reply:
This should be good for now. I'll make an issue for it.